CONTRIBUTING.md

```diff
@@ -0,0 +1,130 @@
+
+Algorithms should:
+
+- have intuitive class and function names that make their purpose clear to readers.
+- use TypScript naming conventions and intuitive variable names to ease comprehension.
+- be flexible to take different input values.
+- raise TypeScript exceptions (RangeError, etc.) on erroneous input values.
+
+Algorithms in this repo should not be how-to examples for existing TypeScript packages. Instead, they should perform
+internal calculations or manipulations to convert input values into different output values. Those calculations or
+manipulations can use data types, classes, or functions of existing TypeScript packages but each algorithm in this repo
+should add unique value.
+
+#### File Naming Convention
+
+- Filenames should use the UpperCamelCase (PascalCase) style.
+- There should be no spaces in filenames.
+- **Example:** `UserProfile.ts` is allowed. Do not use `userprofile.ts`, `Userprofile.ts`, `user-Profile.ts`, `userProfile.ts`; these methods are discouraged.
+
+#### Commit Messages Formatting
+
+- Prefer to use the following format: `<type>: <short summary>`. If necessary, put any extra information in the description.
+- Commit types include (but are not limited to): 
+  - **docs**: Documentantion only changes
+  - **feat**: A new feature
+  - **fix**: A bug fix
+  - **chore**: Miscellaneous stuff that does not match any of the above
+
+- **Examples**: 
+  - `feat: add quicksort algorithm`
+  - `chore: fix spelling`
+  - `fix: improper error message`
+  -  `docs: add contributing guidelines`
+
+#### Module System
+
+We use the [ES Module](https://hacks.mozilla.org/2018/03/es-modules-a-cartoon-deep-dive/) system, which bring an official, standardized module system to JavaScript.
+
+It roughly means you will need to use `export` and `import` statements instead of `module.exports` and `require()`.
+
+#### Coding Style
+
+To maximize the readability and correctness of our code, we require that new submissions follow the
+[TypeScript Standard Style](https://github.com/standard/ts-standard).
+
+In order to apply the coding style (where it can be done automatically). If an error is shown, please figure out what's
+wrong, fix it and run standard again.
+
+A few (but not all) of the things to keep in mind:
+
+- Use camelCase with the leading character as lowercase for identifier names (variables and functions).
+- Names start with a letter.
+- Follow code indentation: Always use 2 spaces for indentation of code blocks.
+
+```ts
+function sum(arr: number[]): number {
+  let total = 0;
+  for (let i = 0; i < arr.length; i++) {
+    total += arr[i];
+  }
+  return total;
+}
+```
+
+- Avoid using global variables and avoid `==`.
+- Please use `let` over `var`.
+- Please refrain from using `console.log` or any other console methods.
+- **Absolutely** don't use `alert`.
+- We strongly recommend the use of ECMAScript 6.
+- Avoid the use of `any` type. Create an interface/class instead.
+- Don't prefix Interfaces with `I`.
```

Comment:
interfaces* and this should perhaps be a more general rule (no redundant naming: don't prefix class members with `m`, functions with `func` or `f` etc.)